// rmwc/next/docs/site-search.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __read = (this && this.__read) || function (o, n) {
    var m = typeof Symbol === "function" && o[Symbol.iterator];
    if (!m) return o;
    var i = m.call(o), r, ar = [], e;
    try {
        while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
    }
    catch (error) { e = { error: error }; }
    finally {
        try {
            if (r && !r.done && (m = i["return"])) m.call(i);
        }
        finally { if (e) throw e.error; }
    }
    return ar;
};
var __spread = (this && this.__spread) || function () {
    for (var ar = [], i = 0; i < arguments.length; i++) ar = ar.concat(__read(arguments[i]));
    return ar;
};
var _this = this;
import React, { useState, useRef } from 'react';
import { menuContent } from './menu-content';
import { Menu, MenuItems, MenuSurfaceAnchor } from '@rmwc/menu';
import { SimpleListItem } from '@rmwc/list';
import { TextField } from '@rmwc/textfield';
import { Link } from 'react-router-dom';
import { CircularProgress } from '@rmwc/circular-progress';
import { history } from './history';
import styles from './site-search.module.css';
var searchComponents = function (val) {
    return menuContent
        .reduce(function (acc, val) {
        if ('options' in val) {
            acc.push.apply(acc, __spread(val.options));
        }
        else {
            acc.push(val);
        }
        return acc;
    }, [])
        .filter(function (c) {
        return c.label.toLowerCase().includes(val.toLowerCase());
    })
        .map(function (c) { return ({
        id: c.label,
        icon: {
            icon: 'code',
            theme: 'primary'
        },
        sectionName: c.label,
        snippet: "View docs for " + c.label,
        url: c.url
    }); });
};
var searchGoogle = function (val, abortController) { return __awaiter(_this, void 0, void 0, function () {
    var _a, items;
    return __generator(this, function (_b) {
        switch (_b.label) {
            case 0: return [4 /*yield*/, fetch("https://www.googleapis.com/customsearch/v1/siterestrict?key=" + process.env.REACT_APP_CUSTOM_SEARCH_KEY + "&cx=" + process.env.REACT_APP_CUSTOM_SEARCH_ID + "&q=" + val, { signal: abortController.signal }).then(function (res) { return res.json(); })];
            case 1:
                _a = (_b.sent()).items, items = _a === void 0 ? [] : _a;
                return [2 /*return*/, (items
                        // shitty hack to ignore things in the index that just reference them homepage
                        .filter(function (r) { return !r.snippet.startsWith('RMWC is a React wrapper'); })
                        .map(function (r) { return ({
                        id: r.cacheId,
                        icon: {
                            icon: 'notes',
                            theme: 'primary'
                        },
                        sectionName: (r.title.split('|').pop() || '').trim(),
                        snippet: r.snippet,
                        // resolve from the full url into the page
                        url: ((r.formattedUrl.split('?').pop() || '')
                            .split('&')
                            .find(function (p) { return p.startsWith('p='); }) || '').slice(2)
                    }); }))];
        }
    });
}); };
export function SiteSearch() {
    var _this = this;
    var _a = __read(useState(''), 2), searchVal = _a[0], _setSearchVal = _a[1];
    var _b = __read(useState(false), 2), isSearching = _b[0], setIsSearching = _b[1];
    var timerIdRef = useRef();
    var abortControllerRef = useRef();
    var _c = __read(useState([]), 2), results = _c[0], setResults = _c[1];
    var doSearch = function (val) {
        if (abortControllerRef.current) {
            abortControllerRef.current.abort();
            abortControllerRef.current = undefined;
        }
        if (timerIdRef.current) {
            clearTimeout(timerIdRef.current);
        }
        var abortController = new AbortController();
        abortControllerRef.current = abortController;
        timerIdRef.current = window.setTimeout(function () { return __awaiter(_this, void 0, void 0, function () {
            var components, searchResults, err_1;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        components = searchComponents(val);
                        setResults(components);
                        _a.label = 1;
                    case 1:
                        _a.trys.push([1, 3, , 4]);
                        return [4 /*yield*/, searchGoogle(val, abortController)];
                    case 2:
                        searchResults = _a.sent();
                        abortControllerRef.current = undefined;
                        setResults(__spread(components, searchResults));
                        return [3 /*break*/, 4];
                    case 3:
                        err_1 = _a.sent();
                        console.warn(err_1);
                        abortControllerRef.current = undefined;
                        return [3 /*break*/, 4];
                    case 4: return [2 /*return*/];
                }
            });
        }); }, 200);
    };
    var setSearchVal = function (val) {
        if (val.length > 2) {
            !isSearching && setIsSearching(true);
            doSearch(val);
        }
        else {
            setResults([]);
        }
        _setSearchVal(val);
    };
    return (React.createElement(MenuSurfaceAnchor, { className: styles.siteSearchWrapper },
        React.createElement(TextField, { placeholder: "Search", icon: "search", trailingIcon: {
                icon: searchVal ? 'close' : '',
                onClick: function () { return setSearchVal(''); }
            }, outlined: true, className: styles.siteSearch, value: searchVal, onChange: function (evt) { return setSearchVal(evt.currentTarget.value); }, onFocus: function () { return setIsSearching(true); }, onKeyDown: function (evt) {
                if (evt.which === 40) {
                    var listItem = document.querySelector("." + styles.siteSearchMenu + " a");
                    if (isSearching && listItem) {
                        listItem.focus();
                    }
                }
            } }),
        React.createElement(Menu, { className: styles.siteSearchMenu, open: isSearching && searchVal.length > 2, anchorCorner: "bottomStart", focusOnOpen: false, onFocus: function () { return setIsSearching(true); }, onClose: function () {
                setIsSearching(false);
            }, onSelect: function (evt) {
                window.scrollTo(0, 0);
                history.replace(evt.detail.item.href.split('/').pop() || '/');
                document.activeElement &&
                    document.activeElement.blur();
            } },
            React.createElement(MenuItems, { twoLine: true },
                results.map(function (r) { return (React.createElement(SimpleListItem, __assign({ tag: Link }, { to: r.url }, { graphic: r.icon, key: r.id, text: r.sectionName, secondaryText: r.snippet }))); }),
                !!abortControllerRef.current && (React.createElement("div", { className: styles.loading },
                    React.createElement(CircularProgress, { size: "large" }))),
                !abortControllerRef.current && !results.length && (React.createElement(SimpleListItem, { text: "No results found" }))))));
}
