// @rmwc/image-list/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var base_1 = require("@rmwc/base");
/** The root of the Image List. */
exports.ImageList = base_1.componentFactory({
    displayName: 'ImageList',
    tag: 'ul',
    classNames: function (props) { return [
        'mdc-image-list',
        {
            'mdc-image-list--masonry': props.masonry,
            'mdc-image-list--with-text-protection': props.withTextProtection
        }
    ]; },
    consumeProps: ['masonry', 'withTextProtection']
});
/** Indicates each item in an Image List. */
exports.ImageListItem = base_1.componentFactory({
    displayName: 'ImageListItem',
    tag: 'li',
    classNames: ['mdc-image-list__item']
});
/** Optional. Parent of each item’s image element, responsible for constraining aspect ratio. This element may be omitted entirely if images are already sized to the correct aspect ratio. */
exports.ImageListImageAspectContainer = base_1.componentFactory({
    displayName: 'ImageListImageAspectContainer',
    classNames: ['mdc-image-list__image-aspect-container']
});
/** Indicates the image element in each item. */
exports.ImageListImage = base_1.componentFactory({
    displayName: 'ImageListImage',
    tag: 'img',
    classNames: ['mdc-image-list__image']
});
/** Optional. Indicates the area within each item containing the supporting text label, if the Image List contains text labels. */
exports.ImageListSupporting = base_1.componentFactory({
    displayName: 'ImageListSupporting',
    classNames: ['mdc-image-list__supporting']
});
/** Optional. Indicates the text label in each item, if the Image List contains text labels. */
exports.ImageListLabel = base_1.componentFactory({
    displayName: 'ImageListLabel',
    classNames: ['mdc-image-list__label']
});
