// rmwc/dist/docs/home.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var react_router_dom_1 = require("react-router-dom");
var __1 = require("../");
var typography_1 = require("@rmwc/typography");
var card_1 = require("@rmwc/card");
var grid_1 = require("@rmwc/grid");
var theme_1 = require("@rmwc/theme");
var icon_1 = require("@rmwc/icon");
var home_module_css_1 = __importDefault(require("./home.module.css"));
var button_1 = require("@rmwc/button");
var menu_content_1 = require("./menu-content");
var checklist = [
    "Uses Google's official material-components-web library",
    'Includes additional addon components not offered by Google.',
    'Works in React from 16.3.x and up',
    'First class Typescript Support',
    'Server side rendering support',
    'Individually packaged and released components'
];
function Example(_a) {
    var ex = _a.ex, name = _a.name, url = _a.url;
    return (React.createElement(react_router_dom_1.Link, { to: url, className: home_module_css_1.default.example, onClick: function () { return window.scrollTo(0, 0); } },
        React.createElement("div", { className: home_module_css_1.default.exampleInner },
            React.createElement("div", null, ex)),
        React.createElement("div", { className: home_module_css_1.default.exampleName },
            React.createElement(typography_1.Typography, { use: "headline6" }, name))));
}
exports.Home = function () {
    return (React.createElement(React.Fragment, null,
        React.createElement("header", { className: home_module_css_1.default.header },
            React.createElement("div", { className: "container" },
                React.createElement(theme_1.Theme, { use: ['onPrimary'] },
                    React.createElement("div", { className: home_module_css_1.default.intro },
                        React.createElement(typography_1.Typography, { use: "headline3", tag: "h1", className: home_module_css_1.default.headline1 }, "React Material Web Components"),
                        React.createElement(typography_1.Typography, { use: "headline6", tag: "h2", className: home_module_css_1.default.headline2 }, "A React UI Kit built on Google's official Material Components Web library"),
                        React.createElement(react_router_dom_1.Link, { to: "/installation" },
                            React.createElement(button_1.Button, { theme: ['secondaryBg', 'onSecondary'], raised: true }, "Get Started")),
                        React.createElement(typography_1.Typography, { tag: "h3", use: "caption" },
                            "v",
                            __1.version),
                        React.createElement("div", { className: home_module_css_1.default.checklist }, checklist.map(function (c) { return (React.createElement("div", { key: c },
                            React.createElement(icon_1.Icon, { theme: "secondary", icon: "check" }),
                            " ",
                            React.createElement("div", null, c))); })))),
                React.createElement(grid_1.Grid, { className: home_module_css_1.default.featureGrid },
                    React.createElement(grid_1.GridCell, { span: 4 },
                        React.createElement(react_router_dom_1.Link, { to: "/installation" },
                            React.createElement(card_1.Card, { theme: ['primaryBg', 'onPrimary'] },
                                React.createElement("div", { style: { padding: '1rem' } },
                                    React.createElement(typography_1.Typography, { use: "headline6", tag: "div" }, "Simple to Use"),
                                    React.createElement(typography_1.Typography, { tag: "div", use: "body1" }, "Get started in 5 minutes.")),
                                React.createElement(card_1.CardMedia, { style: {
                                        backgroundSize: 'fill',
                                        backgroundImage: 'url(images/backgrounds/home-1.png)',
                                        minHeight: '5rem',
                                        flex: 1
                                    } })))),
                    React.createElement(grid_1.GridCell, { span: 4 },
                        React.createElement(card_1.Card, { theme: "background", tag: "a", href: "https://material.io/components/web/" },
                            React.createElement("div", { style: { padding: '1rem' } },
                                React.createElement(typography_1.Typography, { use: "headline6", tag: "div" }, "Material Accuracy"),
                                React.createElement(typography_1.Typography, { tag: "div", use: "body1" }, "Uses Google's official Material Components Web library.")),
                            React.createElement(card_1.CardMedia, { style: {
                                    backgroundSize: 'cover',
                                    backgroundImage: 'url(images/backgrounds/home-2.png)',
                                    minHeight: '5rem',
                                    flex: 1
                                } }))),
                    React.createElement(grid_1.GridCell, { span: 4 },
                        React.createElement(react_router_dom_1.Link, { to: "/styling-theming" },
                            React.createElement(card_1.Card, { style: { backgroundColor: '#212121' }, theme: "textPrimaryOnDark" },
                                React.createElement("div", { style: { padding: '1rem' } },
                                    React.createElement(typography_1.Typography, { use: "headline6", tag: "div" }, "Easy to Customize"),
                                    React.createElement(typography_1.Typography, { tag: "div", use: "body1" }, "Integrate with a variety of CSS workflows.")),
                                React.createElement(card_1.CardMedia, { style: {
                                        backgroundSize: 'cover',
                                        backgroundImage: 'url(images/backgrounds/home-3.png)',
                                        minHeight: '5rem',
                                        flex: 1
                                    } }))))))),
        React.createElement("div", { className: home_module_css_1.default.examples }, menu_content_1.galleryContent.map(function (ex) {
            return (React.createElement(Example, { key: ex.label, name: ex.label, ex: ex.gallery, url: ex.url }));
        }))));
};
exports.default = exports.Home;
